### ISSUE TITLE
IllegalArgumentException: ASTType is null

### ISSUE THREAD

**Issue by danieldietsch** [ultimate-pa/ultimate#291]
Problem: 
"IllegalArgumentException: ASTType is null" during translation.

 * Minimized input: 
[ctrans-bug-min-IAE-asttype_is_null.c](https://github.com/ultimate-pa/ultimate/blob/dev/trunk/examples/CToBoogieTranslation/regression/ctrans-bug-min-IAE-asttype_is_null.c)
 * Settings: ``settings/default/automizer/svcomp-DerefFreeMemtrack-32bit-Automizer_Bitvector.epf``
 * Toolchain: ``toolchains/CheckedCTranslation.xml`` (or any Automizer toolchain) 

This issue occurred during analysis of Trezor firmware and is blocking progress there.

**Comment by invd** [comment]
Unfortunately, there appears to be a regression for this issue for the newest Ultimate 0.1.23-c168100:

```
 * Results from de.uni_freiburg.informatik.ultimate.plugins.generator.cacsl2boogietranslator:
  - ExceptionOrErrorResult: AssertionError: unexpected case
    de.uni_freiburg.informatik.ultimate.plugins.generator.cacsl2boogietranslator: AssertionError: unexpected case: de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.base.CHandler.handleConditionalOperator(CHandler.java:4435)
```

Test settings:
* AutomizerMemDerefMemtrack.xml 
* svcomp-DerefFreeMemtrack-32bit-Automizer_Default.epf

**Comment by alexandernutz** [comment]
thanks for double-checking! -- now I ran _both_ conditional operator examples, so I hope I got them both, finally..

**Comment by invd** [comment]
I can confirm that the "AssertionError: unexpected case" is fixed for Ultimate 0.1.23-277cfe1. Thanks!
